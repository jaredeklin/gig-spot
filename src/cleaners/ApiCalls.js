import { SimpleCleaners } from './SimpleCleaners';
import { cleanConcertData } from './cleanConcertData';

const lastFmApiKey = process.env.REACT_APP_LASTFM_KEY;
const clean = new SimpleCleaners();

export class ApiCalls {
  getLastFmData = concerts => {
    const promises = concerts.map(async concert => {
      const artist = concert.headlineArtist.replace(/&/gi, 'and');
      const baseUrl = 'http://ws.audioscrobbler.com/2.0/?method=artist.getinfo';
      const keyUrl = `&artist=${artist}&api_key=${lastFmApiKey}&format=json`;

      try {
        const response = await fetch(`${baseUrl}${keyUrl}`);
        const artistData = await response.json();
        const image = clean.image(artistData, concert.image);
        const bio = clean.bio(artistData);

        return { ...concert, image, bio };
      } catch (Error) {
        console.log(Error); //eslint-disable-line
        return { ...concert, bio: null };
      }
    });

    return Promise.all(promises);
  };

  getEvents = async (type, url, date) => {
    const dateRange = this.getDateRange(type, date);
    const response = await fetch(`${url}&date=${dateRange}`);

    if (!response.ok) {
      throw Error(response.statusText);
    }

    return await this.cleanData(response);
  };

  cleanData = async response => {
    const concertData = await response.json();
    const cleanData = cleanConcertData(concertData.events.event);

    return await this.getLastFmData(cleanData);
  };

  getDateRange = (type, date) => {
    const { tommorrow, nextWeek, upcoming, upcomingEnd } = date;
    switch (type) {
      case 'today':
        return 'Today';

      case 'week':
        return `${tommorrow}-${nextWeek}`;

      case 'upcoming':
        return `${upcoming}-${upcomingEnd}`;

      default:
        console.log('Invalid type param'); //eslint-disable-line
        break;
    }
  };
}
