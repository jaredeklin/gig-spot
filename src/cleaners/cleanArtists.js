export class CleanArtists {
  clean = (show) => {

    if (!show.performers) {
      const name = show.title.replace(/ [(][0-9][0-9][+] Event[)]/gi, "");
      
      return [name];
    }

    if (show.performers.performer.name) {
      return [show.performers.performer.name];
    }

    return show.performers.performer.map(artist => artist.name);
  };
}