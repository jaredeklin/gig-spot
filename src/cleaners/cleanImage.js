export const cleanImage = (artistData) => {
  if (!artistData.artist || !artistData.artist.image[4]) {

    return null; 
  } else {
    const artistImage = artistData.artist.image[4][`#text`];

    return artistImage;
  }
};