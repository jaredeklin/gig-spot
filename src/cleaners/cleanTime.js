export const cleanTime = time => {
  const startTime = new Date(time).toLocaleTimeString([], {
    hour: '2-digit',
    minute: '2-digit'
  });

  return startTime === '12:00 AM' ? '' : startTime;
};
