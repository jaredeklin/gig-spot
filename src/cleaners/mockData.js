export const mockFetchShowsData = {
    "Info": {
        "TotalResults": 174,
        "PageNumber": 0,
        "Message": null
    },
    "Events": [{
        "Id": 3106726,
        "Date": "2018-04-08T21:00:00",
        "Venue": {
            "Id": 160831,
            "Name": "Streets Of London Pub",
            "Address": "1501 E Colfax Ave",
            "City": "Denver",
            "State": "Colorado",
            "StateCode": "CO",
            "Country": "US",
            "CountryCode": "US",
            "ZipCode": "80218",
            "Url": "",
            "Latitude": 0.0,
            "Longitude": 0.0
        },
        "Artists": [{
            "Id": 123882,
            "Name": "Shawn James"
        }],
        "TicketUrl": "http://www.shareasale.com/r.cfm?u=460319&b=234786&m=27601&afftrack=&urllink=https://www.ticketfly.com/purchase/event/1634017/tfly?utm_medium=api"
    }]
}

export const mockCleanConcertData = [{
        "Id": 3106726,
        "Date": "2018-04-08T21:00:00",
        "Venue": {
            "Id": 160831,
            "Name": "Streets Of London Pub",
            "Address": "1501 E Colfax Ave",
            "City": "Denver",
            "State": "Colorado",
            "StateCode": "CO",
            "Country": "US",
            "CountryCode": "US",
            "ZipCode": "80218",
            "Url": "",
            "Latitude": 0.0,
            "Longitude": 0.0
        },
        "Artists": [{
            "Id": 123882,
            "Name": "Shawn James"
        }],
        "TicketUrl": "http://www.shareasale.com/r.cfm?u=460319&b=234786&m=27601&afftrack=&urllink=https://www.ticketfly.com/purchase/event/1634017/tfly?utm_medium=api"
    }]

export const mockTonightCardData = {
  artist: "Selina Albright", 
  venue: "The Soiled Dove Underground", 
  date: {dayName: "Fri", day: "20", month: "Apr", year: "2018", startTime: "20:00"}, 
  id: 3108932, 
  image: "https://lastfm-img2.akamaized.net/i/u/300x300/2254265bbd13a9aea946244144459709.png" 
};

export const mockTonightCardDataWOImage = {
  artist: "Selina Albright", 
  venue: "The Soiled Dove Underground", 
  date: {dayName: "Fri", day: "20", month: "Apr", year: "2018", startTime: "20:00"}, 
  id: 3108932, 
  image: null
};

export const mockFetchImageConcertData = [{
  artist: "Shawn James", venue: "Streets Of London Pub", date: "Apr 8", startTime: "9:00 PM", id: 3106726
}];

export const mockFetchArtistImageReturnData = {
  artist: {
    bio: {links: {}, published: "18 Sep 2012, 01:00", summary: "Shawn James" },
    image: [
      {'#text': "https://lastfm-img2.akamaized.net/i/u/34s/ef281d679fd4707227ec33aaa633ca95.png", size: "small"},
      {'#text': "https://lastfm-img2.akamaized.net/i/u/64s/ef281d679fd4707227ec33aaa633ca95.png", size: "medium"},
      {'#text': "https://lastfm-img2.akamaized.net/i/u/174s/ef281d679fd4707227ec33aaa633ca95.png", size: "large"},
      {'#text': "https://lastfm-img2.akamaized.net/i/u/300x300/ef281d679fd4707227ec33aaa633ca95.png", size: "extralarge"},
      {'#text': "https://lastfm-img2.akamaized.net/i/u/300x300/ef281d679fd4707227ec33aaa633ca95.png", size: "mega"},
      {'#text': "https://lastfm-img2.akamaized.net/i/u/300x300/ef281d679fd4707227ec33aaa633ca95.png", size: ""}
      ],
    name: "Shawn James",
  }
}

export const mockFetchImageReturnData = [{"artist": "Shawn James", "date": "Apr 8", "id": 3106726, "image": "https://lastfm-img2.akamaized.net/i/u/300x300/ef281d679fd4707227ec33aaa633ca95.png", "startTime": "9:00 PM", "venue": "Streets Of London Pub"}]