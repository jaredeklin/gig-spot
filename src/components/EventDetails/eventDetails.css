.event-details {
  display: grid;
  grid-template-columns: 350px 525px;
  margin: 50px auto;
  width: 80%;
  box-shadow: 1px 0 5px 0;
  max-height: 350px;
  width: 875px;
}

.detail-right {
  grid-column: 2 / 3;
  padding: 30px;
  box-sizing: border-box;
  position: relative;
} 

.detail-img {
  grid-column: 1;
  height: 350px;
  width: 350px;
  opacity: .9;
}

.headliner {
  font-size: 36px;
  width: 350px;
}

.support {
  font-size: 16px;
  margin-top: 5px;
}

.support h4 {
  margin-top: 2px;
}

.detail-date {
  position: absolute;
  top: 0;
  right: 5px;
  height: 100px;
  width: 100px;
  border: 1px solid black;
  border-radius: 50%;
  padding: 20px;
  text-align: center;
  box-sizing: border-box;
  font-size: 24px;
  margin: 30px;
}

.detail-venue {
  position: absolute;
  bottom: 0;
  margin-bottom: 30px;
  width: 330px;
  overflow: scroll;
}

.detail-start-time {
  position: absolute;
  right: 35px;
  bottom: 185px;
  width: 100px;
  text-align: center;
  font-weight: bold;
}

.ticket-info {
  text-align: center;
  position: absolute;
  right: 0;
  bottom: 0;
  margin: 30px;
  height: 50px;
  width: 115px;
  border-left: 1px solid;
  border-right: 1px solid;
}

.ticket-info h4 {
  position: relative;
  top: 15px;
}

.ticket-info:hover {
  box-shadow: 0 0 3px 0;
}

.bio-section {
  width: 875px;
  margin: 0 auto 50px auto;
}

.bio-text {
  margin-top: 20px;
}

@media screen and (max-width: 900px) {

  header {
    background-position-x: -160px; 
  }

  .headliner {
    width: unset;
  }

  .event-details {
    grid-template-columns: 300px;
    grid-template-rows: 300px 80px;
    width: 600px;
    max-height: unset;
  }

  .bio-section {
    width: 600px;
  }

  .detail-img {
    height: 100%;
    width: 300px;
  }

  .detail-venue {
    bottom: -70px;
    left: -300px;
    margin: 0 0 0 10px;
  }

  .ticket-info {
    width: 180px;
    bottom: -95px;
    right: unset;
  }

  .detail-date {
    top: unset;
    right: unset;
    bottom: -20px;
    left: 110px;
  }

  .detail-start-time {
    right: unset;
    bottom: 50px;
    font-weight: bold;
  }

  .landing-page-background {
    background-position-x: center; 
  }
}

@media screen and (max-width: 550px) {
  
  header {
    background-position-x: center;
  }

  .event-details {
    grid-template-columns: 300px;
    grid-template-rows: 300px auto;
    width: 300px;
    margin-top: 25px;
  }

  .bio-section {
    width: 300px;
  }

  .detail-right {
    grid-column: 1;
    grid-row: 2 / 3;
    padding: 10px 15px 150px 15px;
  }

  .headliner {
    width: unset;
    font-size: 24px;
  }

  .ticket-info {
    height: 100px;
    margin: 30px 25px;
    width: 115px;
    right: 0;
    bottom: 0;
  }

  .ticket-info h4 {
    top: 40px;
  }

  .detail-venue {
    position: relative;
    left: 0;
    bottom: 0;
    top: 30px;
    margin: 0 0 30px 0;
    overflow: scroll;
    width: unset;
  }

  .detail-date {
    top: unset;
    right: unset;
    bottom: 0;
    left: 0;
    margin: 30px 20px;
  }

  .detail-start-time {
    display: none;
  }
}