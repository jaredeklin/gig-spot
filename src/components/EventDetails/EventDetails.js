import React from 'react';
import './eventDetails.css';
import PropTypes from 'prop-types';

export const EventDetails = ({ concert }) => { 
  const { 
    headlineArtist, 
    supportArtists, 
    date, 
    image, 
    venue, 
    startTime, 
    tickets
  } = concert;

  const allSupportArtists = supportArtists.map(artist => {
    return (<h4 className="support" key={artist+Date.now()}>{artist}</h4>);
  });

  return <article className="event-details">
    <img src={image} className="detail-img" alt="artist" />
    <div className="detail-right">
      <h2 className="headliner">{headlineArtist}</h2>
      <div className="detail-date">{date}</div>
      {supportArtists.length > 0 && <div className="support">
            With:
        {allSupportArtists}
      </div>}
      <div className="detail-venue">
        <a href={venue.url} target="_blank">
          <h4 className="venue-name">{venue.name}</h4>
        </a>
        <div className="address">
          {venue.address}, {venue.city}
        </div>
      </div>
      <div className="detail-start-time">{startTime}</div>
      {tickets && <a href={tickets} className="ticket-info" target="_blank">
        <h4>Get tickets</h4>
      </a>}
    </div>
  </article>;
};

EventDetails.propTypes = {
  concert: PropTypes.object
};