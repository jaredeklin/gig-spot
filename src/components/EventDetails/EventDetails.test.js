import React from 'react';
import { shallow } from 'enzyme';
import { EventDetails } from './EventDetails';
import { mockConcertProps } from '../../cleaners/mockData';

describe('EventDetails', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<EventDetails concert={mockConcertProps} />);

    expect(wrapper).toMatchSnapshot();
  });
});
