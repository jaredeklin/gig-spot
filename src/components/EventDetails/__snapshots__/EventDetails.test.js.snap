// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventDetails should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    concert={
        Object {
            "date": "Apr 13, 2018",
            "headlineArtist": Object {
              "Id": 34468,
              "Name": "The English Beat",
            },
            "id": 3092703,
            "image": "https://lastfm-img2.akamaized.net/i/u/300x300/fa3db70a62ae49dbbbfbd4183d68289f.png",
            "startTime": "8:00 PM",
            "supportArtists": Array [
              Object {
                "Id": "123882",
                "Name": "Shawn James",
              },
            ],
            "tickets": "http://www.shareasale.com/r.cfm?u=460319&b=234786&m=27601&afftrack=&urllink=https://www.ticketfly.com/purchase/event/1612325/tfly?utm_medium=api",
            "venue": Object {
              "address": "7401 E. 1st Ave",
              "city": "Denver",
              "id": 59319,
              "name": "The Soiled Dove Underground",
              "url": "http://www.soileddove.com/",
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <article
          className="event-details"
>
          <img
                    alt="artist"
                    className="detail-img"
                    onError={[Function]}
                    src="https://lastfm-img2.akamaized.net/i/u/300x300/fa3db70a62ae49dbbbfbd4183d68289f.png"
          />
          <div
                    className="detail-right"
          >
                    <h2
                              className="headliner"
                    >
                              Object {
                                        "Id": 34468,
                                        "Name": "The English Beat",
                                      }
                    </h2>
                    <div
                              className="detail-date"
                    >
                              Apr 13, 2018
                    </div>
                    <div
                              className="support"
                    >
                              With:
                              <h4
                                        className="support"
                              >
                                        Object {
                                                  "Id": "123882",
                                                  "Name": "Shawn James",
                                                }
                              </h4>
                    </div>
                    <div
                              className="detail-venue"
                    >
                              <a
                                        href="http://www.soileddove.com/"
                                        rel="noopener noreferrer"
                                        target="_blank"
                              >
                                        <h4
                                                  className="venue-name"
                                        >
                                                  The Soiled Dove Underground
                                        </h4>
                              </a>
                              <div
                                        className="address"
                              >
                                        7401 E. 1st Ave
                                        , 
                                        Denver
                              </div>
                    </div>
                    <div
                              className="detail-start-time"
                    >
                              8:00 PM
                    </div>
                    <a
                              className="ticket-info"
                              href="http://www.shareasale.com/r.cfm?u=460319&b=234786&m=27601&afftrack=&urllink=https://www.ticketfly.com/purchase/event/1612325/tfly?utm_medium=api"
                              rel="noopener noreferrer"
                              target="_blank"
                    >
                              <h4>
                                        Get tickets
                              </h4>
                    </a>
          </div>
</article>,
        undefined,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="artist"
              className="detail-img"
              onError={[Function]}
              src="https://lastfm-img2.akamaized.net/i/u/300x300/fa3db70a62ae49dbbbfbd4183d68289f.png"
/>,
            <div
              className="detail-right"
>
              <h2
                            className="headliner"
              >
                            Object {
                                          "Id": 34468,
                                          "Name": "The English Beat",
                                        }
              </h2>
              <div
                            className="detail-date"
              >
                            Apr 13, 2018
              </div>
              <div
                            className="support"
              >
                            With:
                            <h4
                                          className="support"
                            >
                                          Object {
                                                        "Id": "123882",
                                                        "Name": "Shawn James",
                                                      }
                            </h4>
              </div>
              <div
                            className="detail-venue"
              >
                            <a
                                          href="http://www.soileddove.com/"
                                          rel="noopener noreferrer"
                                          target="_blank"
                            >
                                          <h4
                                                        className="venue-name"
                                          >
                                                        The Soiled Dove Underground
                                          </h4>
                            </a>
                            <div
                                          className="address"
                            >
                                          7401 E. 1st Ave
                                          , 
                                          Denver
                            </div>
              </div>
              <div
                            className="detail-start-time"
              >
                            8:00 PM
              </div>
              <a
                            className="ticket-info"
                            href="http://www.shareasale.com/r.cfm?u=460319&b=234786&m=27601&afftrack=&urllink=https://www.ticketfly.com/purchase/event/1612325/tfly?utm_medium=api"
                            rel="noopener noreferrer"
                            target="_blank"
              >
                            <h4>
                                          Get tickets
                            </h4>
              </a>
</div>,
          ],
          "className": "event-details",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "artist",
              "className": "detail-img",
              "onError": [Function],
              "src": "https://lastfm-img2.akamaized.net/i/u/300x300/fa3db70a62ae49dbbbfbd4183d68289f.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2
                  className="headliner"
>
                  Object {
                                    "Id": 34468,
                                    "Name": "The English Beat",
                                  }
</h2>,
                <div
                  className="detail-date"
>
                  Apr 13, 2018
</div>,
                <div
                  className="support"
>
                  With:
                  <h4
                                    className="support"
                  >
                                    Object {
                                                      "Id": "123882",
                                                      "Name": "Shawn James",
                                                    }
                  </h4>
</div>,
                <div
                  className="detail-venue"
>
                  <a
                                    href="http://www.soileddove.com/"
                                    rel="noopener noreferrer"
                                    target="_blank"
                  >
                                    <h4
                                                      className="venue-name"
                                    >
                                                      The Soiled Dove Underground
                                    </h4>
                  </a>
                  <div
                                    className="address"
                  >
                                    7401 E. 1st Ave
                                    , 
                                    Denver
                  </div>
</div>,
                <div
                  className="detail-start-time"
>
                  8:00 PM
</div>,
                <a
                  className="ticket-info"
                  href="http://www.shareasale.com/r.cfm?u=460319&b=234786&m=27601&afftrack=&urllink=https://www.ticketfly.com/purchase/event/1612325/tfly?utm_medium=api"
                  rel="noopener noreferrer"
                  target="_blank"
>
                  <h4>
                                    Get tickets
                  </h4>
</a>,
              ],
              "className": "detail-right",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Object {
                    "Id": 34468,
                    "Name": "The English Beat",
                  },
                  "className": "headliner",
                },
                "ref": null,
                "rendered": Object {
                  "Id": 34468,
                  "Name": "The English Beat",
                },
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Apr 13, 2018",
                  "className": "detail-date",
                },
                "ref": null,
                "rendered": "Apr 13, 2018",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "With:",
                    Array [
                      <h4
                        className="support"
>
                        Object {
                                                "Id": "123882",
                                                "Name": "Shawn James",
                                              }
</h4>,
                    ],
                  ],
                  "className": "support",
                },
                "ref": null,
                "rendered": Array [
                  "With:",
                  Object {
                    "instance": null,
                    "key": "[object Object]1540436985197",
                    "nodeType": "host",
                    "props": Object {
                      "children": Object {
                        "Id": "123882",
                        "Name": "Shawn James",
                      },
                      "className": "support",
                    },
                    "ref": null,
                    "rendered": Object {
                      "Id": "123882",
                      "Name": "Shawn James",
                    },
                    "type": "h4",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <a
                      href="http://www.soileddove.com/"
                      rel="noopener noreferrer"
                      target="_blank"
>
                      <h4
                                            className="venue-name"
                      >
                                            The Soiled Dove Underground
                      </h4>
</a>,
                    <div
                      className="address"
>
                      7401 E. 1st Ave
                      , 
                      Denver
</div>,
                  ],
                  "className": "detail-venue",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <h4
                        className="venue-name"
>
                        The Soiled Dove Underground
</h4>,
                      "href": "http://www.soileddove.com/",
                      "rel": "noopener noreferrer",
                      "target": "_blank",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "The Soiled Dove Underground",
                        "className": "venue-name",
                      },
                      "ref": null,
                      "rendered": "The Soiled Dove Underground",
                      "type": "h4",
                    },
                    "type": "a",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "7401 E. 1st Ave",
                        ", ",
                        "Denver",
                      ],
                      "className": "address",
                    },
                    "ref": null,
                    "rendered": Array [
                      "7401 E. 1st Ave",
                      ", ",
                      "Denver",
                    ],
                    "type": "div",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "8:00 PM",
                  "className": "detail-start-time",
                },
                "ref": null,
                "rendered": "8:00 PM",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <h4>
                    Get tickets
</h4>,
                  "className": "ticket-info",
                  "href": "http://www.shareasale.com/r.cfm?u=460319&b=234786&m=27601&afftrack=&urllink=https://www.ticketfly.com/purchase/event/1612325/tfly?utm_medium=api",
                  "rel": "noopener noreferrer",
                  "target": "_blank",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Get tickets",
                  },
                  "ref": null,
                  "rendered": "Get tickets",
                  "type": "h4",
                },
                "type": "a",
              },
            ],
            "type": "div",
          },
        ],
        "type": "article",
      },
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <article
            className="event-details"
>
            <img
                        alt="artist"
                        className="detail-img"
                        onError={[Function]}
                        src="https://lastfm-img2.akamaized.net/i/u/300x300/fa3db70a62ae49dbbbfbd4183d68289f.png"
            />
            <div
                        className="detail-right"
            >
                        <h2
                                    className="headliner"
                        >
                                    Object {
                                                "Id": 34468,
                                                "Name": "The English Beat",
                                              }
                        </h2>
                        <div
                                    className="detail-date"
                        >
                                    Apr 13, 2018
                        </div>
                        <div
                                    className="support"
                        >
                                    With:
                                    <h4
                                                className="support"
                                    >
                                                Object {
                                                            "Id": "123882",
                                                            "Name": "Shawn James",
                                                          }
                                    </h4>
                        </div>
                        <div
                                    className="detail-venue"
                        >
                                    <a
                                                href="http://www.soileddove.com/"
                                                rel="noopener noreferrer"
                                                target="_blank"
                                    >
                                                <h4
                                                            className="venue-name"
                                                >
                                                            The Soiled Dove Underground
                                                </h4>
                                    </a>
                                    <div
                                                className="address"
                                    >
                                                7401 E. 1st Ave
                                                , 
                                                Denver
                                    </div>
                        </div>
                        <div
                                    className="detail-start-time"
                        >
                                    8:00 PM
                        </div>
                        <a
                                    className="ticket-info"
                                    href="http://www.shareasale.com/r.cfm?u=460319&b=234786&m=27601&afftrack=&urllink=https://www.ticketfly.com/purchase/event/1612325/tfly?utm_medium=api"
                                    rel="noopener noreferrer"
                                    target="_blank"
                        >
                                    <h4>
                                                Get tickets
                                    </h4>
                        </a>
            </div>
</article>,
          undefined,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="artist"
                className="detail-img"
                onError={[Function]}
                src="https://lastfm-img2.akamaized.net/i/u/300x300/fa3db70a62ae49dbbbfbd4183d68289f.png"
/>,
              <div
                className="detail-right"
>
                <h2
                                className="headliner"
                >
                                Object {
                                                "Id": 34468,
                                                "Name": "The English Beat",
                                              }
                </h2>
                <div
                                className="detail-date"
                >
                                Apr 13, 2018
                </div>
                <div
                                className="support"
                >
                                With:
                                <h4
                                                className="support"
                                >
                                                Object {
                                                                "Id": "123882",
                                                                "Name": "Shawn James",
                                                              }
                                </h4>
                </div>
                <div
                                className="detail-venue"
                >
                                <a
                                                href="http://www.soileddove.com/"
                                                rel="noopener noreferrer"
                                                target="_blank"
                                >
                                                <h4
                                                                className="venue-name"
                                                >
                                                                The Soiled Dove Underground
                                                </h4>
                                </a>
                                <div
                                                className="address"
                                >
                                                7401 E. 1st Ave
                                                , 
                                                Denver
                                </div>
                </div>
                <div
                                className="detail-start-time"
                >
                                8:00 PM
                </div>
                <a
                                className="ticket-info"
                                href="http://www.shareasale.com/r.cfm?u=460319&b=234786&m=27601&afftrack=&urllink=https://www.ticketfly.com/purchase/event/1612325/tfly?utm_medium=api"
                                rel="noopener noreferrer"
                                target="_blank"
                >
                                <h4>
                                                Get tickets
                                </h4>
                </a>
</div>,
            ],
            "className": "event-details",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "artist",
                "className": "detail-img",
                "onError": [Function],
                "src": "https://lastfm-img2.akamaized.net/i/u/300x300/fa3db70a62ae49dbbbfbd4183d68289f.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h2
                    className="headliner"
>
                    Object {
                                        "Id": 34468,
                                        "Name": "The English Beat",
                                      }
</h2>,
                  <div
                    className="detail-date"
>
                    Apr 13, 2018
</div>,
                  <div
                    className="support"
>
                    With:
                    <h4
                                        className="support"
                    >
                                        Object {
                                                            "Id": "123882",
                                                            "Name": "Shawn James",
                                                          }
                    </h4>
</div>,
                  <div
                    className="detail-venue"
>
                    <a
                                        href="http://www.soileddove.com/"
                                        rel="noopener noreferrer"
                                        target="_blank"
                    >
                                        <h4
                                                            className="venue-name"
                                        >
                                                            The Soiled Dove Underground
                                        </h4>
                    </a>
                    <div
                                        className="address"
                    >
                                        7401 E. 1st Ave
                                        , 
                                        Denver
                    </div>
</div>,
                  <div
                    className="detail-start-time"
>
                    8:00 PM
</div>,
                  <a
                    className="ticket-info"
                    href="http://www.shareasale.com/r.cfm?u=460319&b=234786&m=27601&afftrack=&urllink=https://www.ticketfly.com/purchase/event/1612325/tfly?utm_medium=api"
                    rel="noopener noreferrer"
                    target="_blank"
>
                    <h4>
                                        Get tickets
                    </h4>
</a>,
                ],
                "className": "detail-right",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Object {
                      "Id": 34468,
                      "Name": "The English Beat",
                    },
                    "className": "headliner",
                  },
                  "ref": null,
                  "rendered": Object {
                    "Id": 34468,
                    "Name": "The English Beat",
                  },
                  "type": "h2",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Apr 13, 2018",
                    "className": "detail-date",
                  },
                  "ref": null,
                  "rendered": "Apr 13, 2018",
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "With:",
                      Array [
                        <h4
                          className="support"
>
                          Object {
                                                    "Id": "123882",
                                                    "Name": "Shawn James",
                                                  }
</h4>,
                      ],
                    ],
                    "className": "support",
                  },
                  "ref": null,
                  "rendered": Array [
                    "With:",
                    Object {
                      "instance": null,
                      "key": "[object Object]1540436985197",
                      "nodeType": "host",
                      "props": Object {
                        "children": Object {
                          "Id": "123882",
                          "Name": "Shawn James",
                        },
                        "className": "support",
                      },
                      "ref": null,
                      "rendered": Object {
                        "Id": "123882",
                        "Name": "Shawn James",
                      },
                      "type": "h4",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <a
                        href="http://www.soileddove.com/"
                        rel="noopener noreferrer"
                        target="_blank"
>
                        <h4
                                                className="venue-name"
                        >
                                                The Soiled Dove Underground
                        </h4>
</a>,
                      <div
                        className="address"
>
                        7401 E. 1st Ave
                        , 
                        Denver
</div>,
                    ],
                    "className": "detail-venue",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <h4
                          className="venue-name"
>
                          The Soiled Dove Underground
</h4>,
                        "href": "http://www.soileddove.com/",
                        "rel": "noopener noreferrer",
                        "target": "_blank",
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "The Soiled Dove Underground",
                          "className": "venue-name",
                        },
                        "ref": null,
                        "rendered": "The Soiled Dove Underground",
                        "type": "h4",
                      },
                      "type": "a",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          "7401 E. 1st Ave",
                          ", ",
                          "Denver",
                        ],
                        "className": "address",
                      },
                      "ref": null,
                      "rendered": Array [
                        "7401 E. 1st Ave",
                        ", ",
                        "Denver",
                      ],
                      "type": "div",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "8:00 PM",
                    "className": "detail-start-time",
                  },
                  "ref": null,
                  "rendered": "8:00 PM",
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <h4>
                      Get tickets
</h4>,
                    "className": "ticket-info",
                    "href": "http://www.shareasale.com/r.cfm?u=460319&b=234786&m=27601&afftrack=&urllink=https://www.ticketfly.com/purchase/event/1612325/tfly?utm_medium=api",
                    "rel": "noopener noreferrer",
                    "target": "_blank",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Get tickets",
                    },
                    "ref": null,
                    "rendered": "Get tickets",
                    "type": "h4",
                  },
                  "type": "a",
                },
              ],
              "type": "div",
            },
          ],
          "type": "article",
        },
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
